examples/hybrid/plane/topo_dc_2dinvariant_rhoe.jl

```diff
@@ -0,0 +1,354 @@
+    )
+
+    fω¹ = hcurl.(fw)
+    fω¹ .+= vcurlc2f.(cuₕ)
+
+    # cross product
+    # convert to contravariant
+    # these will need to be modified with topography
+    fu =
+        Geometry.Contravariant13Vector.(Ic2f.(cuₕ)) .+
+        Geometry.Contravariant13Vector.(fw)
+    fu¹ = Geometry.project.(Ref(Geometry.Contravariant1Axis()), fu)
+    fu³ = Geometry.project.(Ref(Geometry.Contravariant3Axis()), fu)
+    @. dw -= fω¹ × fu¹ # Covariant3Vector on faces
+    @. duₕ -= If2c(fω¹ × fu³)
+
+
+    @. duₕ -= hgrad(cp) / cρ
+    vgradc2f = Operators.GradientC2F(
+        bottom = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+        top = Operators.SetGradient(Geometry.Covariant3Vector(0.0)),
+    )
+    @. dw -= vgradc2f(cp) / Ic2f(cρ)
+
+    cE = @. (norm(cuw)^2) / 2 + Φ(z)
+    @. duₕ -= hgrad(cE)
+    @. dw -= vgradc2f(cE)
+
+    # 3) total energy
+
+    @. dρe -= hdiv(cuw * (cρe + cp))
+    @. dρe -= vdivf2c(fw * Ic2f(cρe + cp))
+    @. dρe -= vdivf2c(Ic2f(cuₕ * (cρe + cp)))
+
+    # Uniform 2nd order diffusion
+    ∂c = Operators.GradientF2C()
+    fρ = @. Ic2f(cρ)
+    κ₂ = 75.0 # m^2/s
+
+    ᶠ∇ᵥuₕ = @. vgradc2f(cuₕ.components.data.:1)
+    ᶜ∇ᵥw = @. ∂c(fw.components.data.:1)
+    ᶠ∇ᵥh_tot = @. vgradc2f(h_tot)
+
+    ᶜ∇ₕuₕ = @. hgrad(cuₕ.components.data.:1)
+    ᶠ∇ₕw = @. hgrad(fw.components.data.:1)
+    ᶜ∇ₕh_tot = @. hgrad(h_tot)
+
+    hκ₂∇²uₕ = @. hwdiv(κ₂ * ᶜ∇ₕuₕ)
+    vκ₂∇²uₕ = @. vdivf2c(κ₂ * ᶠ∇ᵥuₕ)
+    hκ₂∇²w = @. hwdiv(κ₂ * ᶠ∇ₕw)
+    vκ₂∇²w = @. vdivc2f(κ₂ * ᶜ∇ᵥw)
+    hκ₂∇²h_tot = @. hwdiv(cρ * κ₂ * ᶜ∇ₕh_tot)
+    vκ₂∇²h_tot = @. vdivf2c(fρ * κ₂ * ᶠ∇ᵥh_tot)
+
+    dfw = dY.w.components.data.:1
+    dcu = dY.uₕ.components.data.:1
+
+    # Laplacian Diffusion (Uniform)
+    @. dcu += hκ₂∇²uₕ
+    @. dcu += vκ₂∇²uₕ
+    @. dfw += hκ₂∇²w
+    @. dfw += vκ₂∇²w
+    @. dρe += hκ₂∇²h_tot
+    @. dρe += vκ₂∇²h_tot
+
+    Spaces.weighted_dss!(dY.Yc)
+    Spaces.weighted_dss!(dY.uₕ)
+    Spaces.weighted_dss!(dY.w)
+
+    return dY
+end
+
+dYdt = similar(Y);
+rhs_invariant!(dYdt, Y, nothing, 0.0);
+
+# run!
+using OrdinaryDiffEq
+timeend = 900.0
+Δt = 0.3
+prob = ODEProblem(rhs_invariant!, Y, (0.0, timeend))
+integrator = OrdinaryDiffEq.init(
+    prob,
+    SSPRK33(),
+    dt = Δt,
+    saveat = 10.0,
+    progress = true,
+    progress_message = (dt, u, p, t) -> t,
+);
+
+if haskey(ENV, "CI_PERF_SKIP_RUN") # for performance analysis
+    throw(:exit_profile)
+end
+
+sol = @timev OrdinaryDiffEq.solve!(integrator)
+
+ENV["GKSwstype"] = "nul"
+import Plots, ClimaCorePlots
+Plots.GRBackend()
+
+dir = "dc_invariant_etot"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+anim = Plots.@animate for u in sol.u
+    Plots.plot(u.Yc.ρe ./ u.Yc.ρ)
+end
+Plots.mp4(anim, joinpath(path, "total_energy.mp4"), fps = 20)
+
+If2c = Operators.InterpolateF2C()
+anim = Plots.@animate for u in sol.u
+    Plots.plot(Geometry.WVector.(Geometry.Covariant13Vector.(If2c.(u.w))))
```

Comment:
The `rhs!` evaluation runs now, but the code fails at this point now; I'm not sure why adding the `weight` to the dss term should change this behaviour

Reply:
```[ Info: Precompiling ClimaCorePlots [cf7c7e5a-b407-4c48-9047-11a94a308626]
┌ Info: Saved animation to
└   fn = "/Users/asridhar/Research/Codes/ClimaCore.jl/examples/hybrid/plane/output/dc_invariant_etot/total_energy.mp4"
ERROR: LoadError: InexactError: transform(ClimaCore.Geometry.LocalAxis{(3,)}, [-2.563510965609378e-21, -4.1232239390994916e-5])```

Reply:
The `hardcoded zero slope` fix still works with the latest main branch merged